### PR TITLE
Create OpenAPIV3 schema for kapp-controller package and generate package with the schema

### LINKED ISSUE #2766: Create OpenAPIV3 schema for kapp-controller in tce
## Feature Request

Create OpenAPIV3 schema for kapp-controller in tce

### PR DESCRIPTION
**Pull request by shivaani0505**
## What this PR does / why we need it

- [x] Create OpenAPIV3 schema for kapp-controller package and generate package with the schema (0.30.0)
- [x] Fix bug in package-overlay.yaml
- [x] Add checks to scripts to error out when ytt manifests not generated successfully

## Details for the Release Notes (PLEASE PROVIDE)

```release-note
Create OpenAPIV3 schema for kapp-controller package and generate package with the schema 
```

## Which issue(s) this PR fixes

Fixes: #2766 

## Describe testing done for PR

Tested with following ytt commands locally:
* make generate-openapischema-package PACKAGE=kapp-controller VERSION=0.30.0
* make push-package PACKAGE=kapp-controller VERSION=0.30.0 TAG=0.30.0

## Special notes for your reviewer

ytt version => 0.38.0
kapp version => 0.37.0
yq version => 4.11.1

### CODE REVIEW

File: addons/packages/kapp-controller/0.30.0/bundle/config/schema.yaml [hunk 1]
```diff
@@ -0,0 +1,45 @@
+#! schema.yaml
+
+#@data/values-schema
+#@schema/desc "OpenAPIv3 Schema for kapp-controller"
+---
+#@schema/desc "The namespace value used by older templates, will overwrite with top level namespace of present, for backward compatibility"
```
Comment (maralavi) [inline addons/packages/kapp-controller/0.30.0/bundle/config/schema.yaml:6]: typo "of present"--> "if present"

File: addons/packages/kapp-controller/0.30.0/bundle/config/schema.yaml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+namespace: kapp-controller
+#@schema/desc "Configuration for kapp-controller"
+kappController:
+  #@schema/desc "The namespace value used by older templates, will overwrite with top level namespace of present, for backward compatibility"
```
Comment (maralavi) [inline addons/packages/kapp-controller/0.30.0/bundle/config/schema.yaml:10]: typo "of present"--> "if present"

File: addons/packages/kapp-controller/0.30.0/bundle/config/schema.yaml [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  #@schema/nullable
+  namespace: kapp-controller
```
Comment (maralavi) [inline addons/packages/kapp-controller/0.30.0/bundle/config/schema.yaml:12]: Do both this "namespace" field and the one above in line 7, have same descriptions?

File: addons/packages/kapp-controller/0.30.0/bundle/config/schema.yaml [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+  #@schema/desc "The namespace value used for global packaging resources. Any Package and PackageMetadata CRs within that namespace will be included in all other namespaces on the cluster, without duplicating them"
+  globalNamespace: tanzu-package-repo-global
+  deployment:
+    #@schema/desc "The coreDNSIP will be injected into /etc/resolv.conf of kapp-controller pod"
+    #@schema/nullable
+    coreDNSIP: ""
+    #@schema/desc "HostNetwork of kapp-controller deployment"
+    #@schema/nullable
+    hostNetwork: ""
+    #@schema/desc "PriorityClassName of kapp-controller deployment"
```
Comment (maralavi) [inline addons/packages/kapp-controller/0.30.0/bundle/config/schema.yaml:24]: Do such fields such as "hostNetwork" or "hostNetwork" or alike, need to be more descriptive, e.g. explaining in high level, what is "PriorityClassName"?

File: hack/packages/verify-openapischema-for-package.sh [hunk 5]
```diff
@@ -40,10 +40,16 @@ verify_openapischema_for_package() {
...
-    yq e '.components.schemas.dataValues' generated-openapi-schema.yaml > schema-contents.yaml
-    yq e '.spec.valuesSchema.openAPIv3' ../package.yaml > package-schema-contents.yaml
-    diffyaml schema-contents.yaml package-schema-contents.yaml
-    echo -e "${GREEN}===> OpenAPIv3 contents match successful for schema and package${NC}"
+    status=$?
+    if [ $status -eq 0 ]; then
+      yq e '.components.schemas.dataValues' generated-openapi-schema.yaml > schema-contents.yaml
+      yq e '.spec.valuesSchema.openAPIv3' ../package.yaml > package-schema-contents.yaml
+      diffyaml schema-contents.yaml package-schema-contents.yaml
+      echo -e "${GREEN}===> OpenAPIv3 contents match successful for schema and package${NC}"
```
Comment (maralavi) [inline hack/packages/verify-openapischema-for-package.sh:48]: suggestion: wouldn't "successfully matched" be better?

File: addons/packages/kapp-controller/0.30.0/bundle/config/schema.yaml (hunk 3 above)
Comment (shivaani0505) [reply addons/packages/kapp-controller/0.30.0/bundle/config/schema.yaml:12]: no, it shud be different. Fixed

File: addons/packages/kapp-controller/0.30.0/bundle/config/schema.yaml (hunk 2 above)
Comment (shivaani0505) [reply addons/packages/kapp-controller/0.30.0/bundle/config/schema.yaml:10]: Fixed

File: addons/packages/kapp-controller/0.30.0/bundle/config/schema.yaml (hunk 1 above)
Comment (shivaani0505) [reply addons/packages/kapp-controller/0.30.0/bundle/config/schema.yaml:6]: Fixed

File: hack/packages/verify-openapischema-for-package.sh (hunk 5 above)
Comment (shivaani0505) [reply hack/packages/verify-openapischema-for-package.sh:48]: Fixed

File: addons/packages/kapp-controller/0.30.0/bundle/config/schema.yaml (hunk 4 above)
Comment (shivaani0505) [reply addons/packages/kapp-controller/0.30.0/bundle/config/schema.yaml:24]: For priorityClassName, I can add additional line => "This value references an existing Kubernetes priorityClassName that can be assigned to kapp-controller pods"
What do you think?

Comment (maralavi) [reply addons/packages/kapp-controller/0.30.0/bundle/config/schema.yaml:24]: Yes sounds good, or the following from kapp controller in [here](https://github.com/vmware-tanzu/carvel-kapp-controller/blob/develop/vendor/k8s.io/api/core/v1/types.go): "The priority value that various system components use to find the priority of the pod"

Comment (shivaani0505) [reply addons/packages/kapp-controller/0.30.0/bundle/config/schema.yaml]: Updated the descriptions

**Review by maralavi — DISMISSED** [review]

File: addons/packages/kapp-controller/0.30.0/bundle/config/schema.yaml [hunk 6]
```diff
@@ -18,10 +18,10 @@ kappController:
     #@schema/desc "The coreDNSIP will be injected into /etc/resolv.conf of kapp-controller pod"
     #@schema/nullable
     coreDNSIP: ""
-    #@schema/desc "HostNetwork of kapp-controller deployment"
+    #@schema/desc "Host networking requested for kapp-controller pod. Use the host's network namespace. If this option is set, the ports that will be used must be specified"
```
Comment (maralavi) [inline addons/packages/kapp-controller/0.30.0/bundle/config/schema.yaml:21]: thought: How about just the first sentence "Host networking requested for kapp-controller pod" and omitting the other sentences after that, as we don't have any instructions how to set ports

Comment (shivaani0505) [reply addons/packages/kapp-controller/0.30.0/bundle/config/schema.yaml]: Fixed.

**Review by maralavi — APPROVED** [review]

**Review by seemiller — APPROVED** [review]

**Pull request merged**